src/darray.jl

```diff
@@ -90,6 +98,7 @@ function DArray(id, init, dims, pids, idxs, cuts)
     A = take!(r)
     if myid() in pids
         d = registry[id]
+        d = isa(d, WeakRef) ? d.value : d
```

Comment:
Maybe we should define `getindex(w::WeakRef)` in Base such that we in the future can just write `d[]` in places like this.